### ISSUE TITLE
PortMidi backend returns Invalid device ID when opening a port

### ISSUE THREAD

**Issue by jacobanana** [mido/mido#169]
Hello,

I have recently updated from version 1.1.18 where portmidi was the default backend to 1.2.8 and I am now unable to use it as it throws the following error.

```
>>> import mido
>>> mido.set_backend('mido.backends.portmidi')
>>> inputs = mido.get_input_names()
>>> inputs
[u'Launchpad Pro MIDI 1', u'Launchpad Pro MIDI 2', u'Launchpad Pro MIDI 3', u'Midi Through Port-0']
>>> port = mido.open_input(inputs[0])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/adrien/test/venv/local/lib/python2.7/site-packages/mido/backends/backend.py", line 90, in open_input
    return self.module.Input(name, **self._add_api(kwargs))
  File "/home/adrien/test/venv/local/lib/python2.7/site-packages/mido/ports.py", line 164, in __init__
    BasePort.__init__(self, name, **kwargs)
  File "/home/adrien/test/venv/local/lib/python2.7/site-packages/mido/ports.py", line 88, in __init__
    self._open(**kwargs)
  File "/home/adrien/test/venv/local/lib/python2.7/site-packages/mido/backends/portmidi.py", line 129, in _open
    pm.null))      # Time info
  File "/home/adrien/test/venv/local/lib/python2.7/site-packages/mido/backends/portmidi.py", line 34, in _check_error
    raise IOError(pm.lib.Pm_GetErrorText(return_value))
IOError: PortMidi: `Invalid device ID'
```

*rtmidi* and *pygame* (based on portmidi) work fine.

I tried reverting versions and it appears that this got introduced at version 1.2.1. Version 1.2.0 also doesn't work but throws a different error which is not related:

```
AttributeError: module 'mido.backends.portmidi_init' has no attribute 'byref'
```

The reason for not wanting to use RTMidi is because of an issue on OS-X Sierra and later : https://github.com/thestk/rtmidi/issues/155

Conditions for repro:
OS-X 10.13.5 or Raspbian Jessie
Python 2.7 or 3.x
Mido 1.2.1 or later
PortMidi 217 (installed with homebrew on os-x)

**Comment by radarsat1** [comment]
If anyone can help debug this for rtmidi I'm all ears. Is there a MIDI driver problem in OS X High Sierra?  (I don't have that OS so I can't test easily..)

**Comment by dagargo** [comment]
I had the same issue but looks like it got fixed in 1.2.9, commit d723514f7626afeefd29d4db40a0444ee300d3ea.
Could someone else confirm this?

**Comment by rdoursenaud** [comment]
I'll consider the issue resolved per @dagargo comment.
Feel free to reopen if not.
